illumos-utils/src/lib.rs

```diff
@@ -92,6 +92,31 @@ mod inner {
 
     // Helper function for starting the process and checking the
     // exit code result.
+    pub fn spawn(
+        command: &mut std::process::Command,
+    ) -> Result<std::process::Child, ExecutionError> {
+        let child = command.spawn().map_err(|err| {
+            ExecutionError::ExecutionStart { command: to_string(command), err }
+        })?;
+
+        Ok(child)
```

Comment:
Style nit, but I think this can just be written as:

```rust
command.spawn().map_err(|err| {
    ExecutionError::ExecutionStart { command: to_string(command), err }
})
```

I.e, no need to use the `?`, we can just map the error type.